### ISSUE TITLE
File extraction fails when give filename str as a `targets` argument in v0.21.0

### ISSUE THREAD

**Issue by huettenhain** [miurahr/py7zr#576]
Since updating to 0.21, the following happens:
```
Python 3.12.1 (tags/v3.12.1:2305ca5, Dec  7 2023, 22:03:25) [MSC v.1937 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import base64
>>> from py7zr import SevenZipFile
>>> from io import BytesIO
>>> data = base64.b64decode(
...     'N3q8ryccAAT9xtacpQAAAAAAAAAiAAAAAAAAAEerl+XBRlkrcJwwoqgijCyuEh0S'
...     'qLfjamv2F2vNJGFGyHDfpAAAgTMHrg/QDrA8nzkQnJ+m1TPasi6xAvSHzZaZrISL'
...     'D+EsvFULZ44Kf7Ewy47PApbKruXCaOSUsjzeqpG8VBcx66h2cV/lnGDfUjtVsyGB'
...     'HmmmTaSI/atXtuwiN5mGrqyFZTC/V2VEohWua1Yk1K+jXy+32hBwnK2clyr3rN5L'
...     'Abv5g2wXBiABCYCFAAcLAQABIwMBAQVdABAAAAyAlgoBouB4BAAA')
>>> arc = SevenZipFile(BytesIO(data), password='boom')
>>> arc.read('bar.txt') 
{}
>>> arc.read()
{'bar.txt': <_io.BytesIO object at 0x000001ECFC3A2020>, 'foo.txt': <_io.BytesIO object at 0x000001ECFEA97BA0>}
>>> arc.reset()
>>> arc.read('bar.txt') 
{}
>>> arc.read()['bar.txt'].read()
b'refinery'
```
The expected behaviour would be that a call to `arc.read('bar.txt')` returns the same reader to the contents of `bar.txt` that is returned by `arc.read()`.

**Comment by miurahr** [comment]
Does previous version works as intended?

**Comment by miurahr** [comment]
There is a wrong argument for read method.

> arc.read('bar.txt')

read accept list of str as argument, but you give str.
```python
    def read(self, targets: Optional[List[str]] = None) -> Optional[Dict[str, IO[Any]]]:
```

It should be `arc.read(['bar'])`

**Comment by miurahr** [comment]
It seems a common mistake to give `str` instead of `List[str]` for argument. A behavior is undefined when giving `str` for argument. Is it better to raise an exception when accepting a wrong argument?

**Comment by miurahr** [comment]
Here is an idea to check type and raise error when given other than list

```python
    def read(self, targets: Optional[Collection[str]] = None) -> Optional[Dict[str, IO[Any]]]:
        if targets is not None and not isinstance(targets, Collection):
            raise TypeError("Wrong argument type given.")
```

It pass the test case as follows:

```python
    with py7zr.SevenZipFile(BytesIO(data), password="boom") as arc:
        result = arc.read(["bar.txt"])
        assert result.get("bar.txt").read() == b"refinery"
    with py7zr.SevenZipFile(BytesIO(data), password="boom") as arc:
        result = arc.read({"bar.txt"})
        assert result.get("bar.txt").read() == b"refinery"
    with pytest.raises(TypeError):
        with py7zr.SevenZipFile(BytesIO(data), password="boom") as arc:
            arc.read("bar.txt")
    with pytest.raises(TypeError):
        with py7zr.SevenZipFile(BytesIO(data), password="boom") as arc:
            arc.read("bar.txt")

```

There should be consistent among methods like `read` and  `extract`

**Comment by miurahr** [comment]
> But I must apologize, the type hint is actually quite clear. I think a `TypeError` is a good solution, that would have likely made me realize my mistake earlier. I'll note that your current check would match `list | set | None` rather than `Collection[str] | None`. For the latter, I would suggest the following type check:

I think a type check should be minimum and a check for items of collection is overkill in a common sense of python culture, and we have already provide a type hint.
I think it is enough to check a common mistake,  give str object as argument and it is set or list.

**Comment by huettenhain** [comment]
Your code, your rules. I got everything I needed, thanks a lot! 😃

**Comment by miurahr** [comment]
I've merged #577. Now we can close here.
